src/test/java/ro/unibuc/hello/controller/GreetingsControllerIntegrationTest.java

```diff
@@ -0,0 +1,139 @@
+    public void cleanUpAndAddTestData() {
+        greetingsService.deleteAllGreetings();
+        
+        Greeting greeting1 = new Greeting("1", "Hello 1");
+        Greeting greeting2 = new Greeting("2", "Hello 2");
+
+        greetingsService.saveGreeting(greeting1);
+        greetingsService.saveGreeting(greeting2);
+    }
+
+    @Test
+    public void testGetAllGreetings() throws Exception {
+        mockMvc.perform(get("/greetings"))
+            .andExpect(status().isOk())
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(jsonPath("$.length()").value(2))
+            .andExpect(jsonPath("$[0].content").value("Hello 1"))
+            .andExpect(jsonPath("$[1].content").value("Hello 2"));
+    }
+
+    @Test
+    public void testCreateGreeting() throws Exception {
+        Greeting greeting = new Greeting("3", "Hello New");
+
+        mockMvc.perform(post("/greetings")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(new ObjectMapper().writeValueAsString(greeting)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value("3"))
+                .andExpect(jsonPath("$.content").value("Hello New"));
+
+        mockMvc.perform(get("/greetings"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.length()").value(3));
+    }
+
+    @Test
+    public void testUpdateGreeting() throws Exception {
+        Greeting greeting = new Greeting("1", "Hello Updated");
+
+        mockMvc.perform(put("/greetings/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(new ObjectMapper().writeValueAsString(greeting)))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.id").value("1"))
+                .andExpect(jsonPath("$.content").value("Hello Updated"));
+
+        mockMvc.perform(get("/greetings"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.length()").value(2))
+                .andExpect(jsonPath("$[0].content").value("Hello Updated"))
+                .andExpect(jsonPath("$[1].content").value("Hello 2"));
+    }
+
+    @Test
+    public void testDeleteGreeting() throws Exception {
+
+        mockMvc.perform(delete("/greetings/1"))
+            .andExpect(status().isOk());
+
+        mockMvc.perform(get("/greetings"))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andExpect(jsonPath("$[0].content").value("Hello 2"));
+    }
+}
```

Comment:
Nit: no new line at end of file, there are a few more files with this issue

Reply:
@alexOporanu1, addressed in 653d7df000551cd920d156103c5c4fc9760f5d84, thank you.